python/mlc_chat/interface/openai_api.py

```diff
@@ -28,61 +29,61 @@ class ChatCompletionRequest(BaseModel):
 
 class UsageInfo(BaseModel):
     prompt_tokens: int = 0
-    completion_tokens: Optional[int] = 0
+    completion_tokens: int | None = 0
     total_tokens: int = 0
 
 class ChatCompletionResponseChoice(BaseModel):
     index: int
     message: ChatMessage
-    finish_reason: Optional[Literal["stop", "length"]]
+    finish_reason: Literal["stop", "length"] | None = None
 
 class ChatCompletionResponse(BaseModel):
     id: str = Field(default_factory=lambda: f"chatcmpl-{shortuuid.random()}")
     object: str = "chat.completion"
     created: int = Field(default_factory=lambda: int(time.time()))
-    choices: List[ChatCompletionResponseChoice]
+    choices: list[ChatCompletionResponseChoice]
     # TODO: Implement support for the following fields
-    usage: Optional[UsageInfo]
+    usage: UsageInfo | None = None
 
 class DeltaMessage(BaseModel):
-    role: Optional[str] = None
-    content: Optional[str] = None
+    role: str | None = None
+    content: str | None = None
 
 class ChatCompletionResponseStreamChoice(BaseModel):
     index: int
     delta: DeltaMessage
-    finish_reason: Optional[Literal["stop", "length"]]
+    finish_reason: Literal["stop", "length"] | None = None
 
 class ChatCompletionStreamResponse(BaseModel):
     id: str = Field(default_factory=lambda: f"chatcmpl-{shortuuid.random()}")
     object: str = "chat.completion.chunk"
     created: int = Field(default_factory=lambda: int(time.time()))
-    choices: List[ChatCompletionResponseStreamChoice]
+    choices: list[ChatCompletionResponseStreamChoice]
 
 class CompletionRequest(BaseModel):
     model: str
-    prompt: Union[str, List[Any]]
+    prompt: str | list[str]
 
 class CompletionResponseChoice(BaseModel):
     index: int
     text: str
-    logprobs: Optional[int] = None
-    finish_reason: Optional[Literal["stop", "length"]]
+    logprobs: int | None = None
+    finish_reason: Literal["stop", "length"] | None = None
 
 class CompletionResponse(BaseModel):
     id: str = Field(default_factory=lambda: f"cmpl-{shortuuid.random()}")
     object: str = "text_completion"
     created: int = Field(default_factory=lambda: int(time.time()))
-    choices: List[CompletionResponseChoice]
+    choices: list[CompletionResponseChoice]
     usage: UsageInfo
 
 class EmbeddingsRequest(BaseModel):
-    model: Optional[str] = None
-    input: Union[str, List[Any]]
-    user: Optional[str] = None
+    model: str
+    input: str
+    user: str | None = None
 
 class EmbeddingsResponse(BaseModel):
     object: str = "list"
-    data: List[Dict[str, Any]]
-    model: Optional[str] = None
+    data: list[Dict[str, Any]]
```

Comment:
It looks like not the type-annotation style used in MLC-LLM and TVM. Could you explain the reason you'd like to change the style?

Reply:
i think lower-cased `list` is pretty much a newer python feature since python 3.9